src/server/engine_shard_set.cc

```diff
@@ -872,7 +874,7 @@ ShardId Shard(string_view v, ShardId shard_num) {
 
   XXH64_hash_t hash = XXH64(v.data(), v.size(), 120577240643ULL);
 
-  if (has_hashtags) {
+  if (has_hashtags || RoundRobinSharder::IsEnabled()) {
```

Comment:
you dont need the has_hashtags condition here
TryGetShardId will return nullopt if !IsEnabled() you can remove this check from TryGetShardId 
and in this if condition just use RoundRobinSharder::IsEnabled()